src/Microsoft.ComponentDetection.Detectors/rust/RustCliDetector.cs

```diff
@@ -120,6 +171,27 @@ private static (string Name, string Version) ParseNameAndVersion(string nameAndV
         return (parts[0], parts[1]);
     }
 
+    private static bool IsLocalPackage(CargoPackage package) => package.Source == null;
+
+    private static bool ParseDependency(string dependency, out string packageName, out string version, out string source)
+    {
+        var match = DependencyFormatRegex.Match(dependency);
+        var packageNameMatch = match.Groups["packageName"];
+        var versionMatch = match.Groups["version"];
+        var sourceMatch = match.Groups["source"];
+
+        packageName = packageNameMatch.Success ? packageNameMatch.Value : null;
```

Comment:
Is `.Value` not null if `Success` is false?

Reply:
Only if none of the groups match, so in the case of it just being a package name the other two fields have a value of "" instead of null. 

But if none of them match then yes they are all null 👎